0.71.0 tests failed on Fedora 30

test_report:
[test_report_0.71.0_f30.txt](https://github.com/zim-desktop-wiki/zim-desktop-wiki/files/3120569/test_report_0.71.0_f30.txt)

Errors reported by @cookiEoverdose still happen with the current master. However, they are unrelated to Fedora buildroot, and instead caused by language settings. For instance:

> AssertionError: 'Anscheinend haben Sie einen Fehler gefunden' != 'Looks like you found a bug'
> - Anscheinend haben Sie einen Fehler gefunden
> + Looks like you found a bug

 It seems that the tests must be run in English locale to succeed. This command works: `LANG=en_US.UTF-8 ./test.py`

--- a/tests/__init__.py
+++ b/tests/__init__.py
@@ -118,7 +118,8 @@ def _setUpEnvironment():
 		'TEST_XDG_DATA_DIRS': os.path.join(TMPDIR, 'data_dir'),
 		'XDG_CONFIG_HOME': os.path.join(TMPDIR, 'config_home'),
 		'XDG_CONFIG_DIRS': os.path.join(TMPDIR, 'config_dir'),
-		'XDG_CACHE_HOME': os.path.join(TMPDIR, 'cache_home')
+		'XDG_CACHE_HOME': os.path.join(TMPDIR, 'cache_home'),
+		'XDG_RUNTIME_DIR': os.path.join(TMPDIR, 'runtime')
 	})
 
 	if os.path.isdir(TMPDIR):

--- a/tests/newfs.py
+++ b/tests/newfs.py
@@ -104,7 +104,8 @@ class TestFilePath(tests.TestCase):
 
 		# Test home folder fallback
 		f = FilePath('~non-existing-user/foo')
-		self.assertEqual(f.path, P('/'.join((HOME.dirname, 'non-existing-user', 'foo'))))
+		homedirname = HOME.dirname or '' # Build environments may have HOME set as e.g. '/builddir'
+		self.assertEqual(f.path, P('/'.join((homedirname, 'non-existing-user', 'foo'))))
 
 
 	def testShareDrivePath(self):
@@ -179,7 +180,9 @@ class TestFilePath(tests.TestCase):
 			self.assertEqual(f.get_abspath(p).path, want)
 
 	def testUserpath(self):
-		self.assertTrue(len(HOME.pathnames) >= 2)
+		# Not always true, e.g. a build environment may have home set to
+		# /builddir.
+		#self.assertTrue(len(HOME.pathnames) >= 2)
 
 		f = FilePath('~/foo')
 		self.assertEqual(f.path, HOME.get_childpath('foo').path)

--- a/tests/uiactions.py
+++ b/tests/uiactions.py
@@ -22,6 +22,10 @@ import zim.gui
 
 from zim.gui.uiactions import UIActions, PAGE_EDIT_ACTIONS
 
+def strip_file_url_prefix(str):
+	if str.startswith('file://'): return str[7:]
+	return str
+
 
 class EmptyWindowObject(object):
 	# Any other method than get_toplevel() should raise to prevent
@@ -853,7 +857,7 @@ class TestUIActionsRealFile(tests.TestCase):
 		folder.touch()
 
 		def open_folder(cmd):
-			self.assertEqual(cmd[-1], folder.path)
+			self.assertEqual(strip_file_url_prefix(cmd[-1]), folder.path)
 
 		with tests.ApplicationContext(open_folder):
 			with tests.DialogContext():
@@ -867,7 +871,7 @@ class TestUIActionsRealFile(tests.TestCase):
 			dialog.answer_yes()
 
 		def open_folder(cmd):
-			self.assertEqual(cmd[-1], folder.path)
+			self.assertEqual(strip_file_url_prefix(cmd[-1]), folder.path)
 
 		with tests.ApplicationContext(open_folder):
 			with tests.DialogContext(create_folder):
@@ -875,7 +879,7 @@ class TestUIActionsRealFile(tests.TestCase):
 
 	def testOpenNotebookFolder(self):
 		def open_folder(cmd):
-			self.assertEqual(cmd[-1], self.notebook.folder.path)
+			self.assertEqual(strip_file_url_prefix(cmd[-1]), self.notebook.folder.path)
 
 		with tests.ApplicationContext(open_folder):
 			self.uiactions.open_notebook_folder()
@@ -886,7 +890,7 @@ class TestUIActionsRealFile(tests.TestCase):
 		self.notebook.document_root.touch()
 
 		def open_folder(cmd):
-			self.assertEqual(cmd[-1], self.notebook.document_root.path)
+			self.assertEqual(strip_file_url_prefix(cmd[-1]), self.notebook.document_root.path)
 
 		with tests.ApplicationContext(open_folder):
 			with tests.DialogContext():

